### ISSUE TITLE
Not compiling, needs an older libmbedcrypto 

### ISSUE THREAD

**Issue by HenriLaurie** [rafaqz/Microclimate.jl#4]
Very keen to try this, been hoping to find it and should have done so sooner. Pity it is  currently somewhat inactive---but I want to try it on a course (4th year applied maths, module on mathematical biology, includes a DEB section). Might be back with more queries!

In any case, I can compile the other two packages (DynamicEnergyBudgets, Photosynthesis) but not this one, the important lines in the error are
```
ERROR: LoadError: InitError: could not load library "/home/henri/.julia/artifacts/0a7f440ba143b238cc3dd0def2956d34d76755f3/lib/libnetcdf.so"
libmbedcrypto.so.5: cannot open shared object file: No such file or directory

```

I'm using an up-to-date Manjaro, there is an Arch error out there at 
https://bugs.archlinux.org/task/70173
which says Julia 1.6 can't find libmbedcrypto.so.5 because it uses libmbedcrypto.so.6. The workaround there seems to me not to apply here.

**Comment by rafaqz** [comment]
Hi! thanks for the interest in this. Its inactive because I have since written spatial data handling more generally in GeoData.jl and RasterDataSources.jl, see https://github.com/EcoJulia/RasterDataSources.jl/issues/17. They're largely inspired by the fact that I had to write this package at all.

And I had to write too much tooling from scratch for the DEB paper, it was just a masters project. There just wasn't anything to build on. But it turned out to be specific for me to maintain when I'm not actually working with DEB/Microclim currently. 

But we should be able to do much the same thing with GeoData.jl now. It's just been a lot of work to get to doing it properly.

**Comment by HenriLaurie** [comment]
So I have looked under the hood a little bit. For my immediate purpose it would be enough to run some rudimentary simulations of an imaginary animal. I see that organism.jl (I assume there's only one such file, but if not I am talking about the one in src of the package DynamicEnergyBudgets) defines only one subtype of AbstractOrganism, namely Plant, and that this was "due to current limitations of Julia" which Julia 1.5 would have overcome, so in principle I could just fork and tinker a bit to make an example run for my students.

Of course I could just present them with the standard model implemented as a set of ODEs, with Julia code to solve it. But I would love for the code here to cover the whole of DEB and eventually become a candidate to replace the Matlab code of DEBtool. I see Matlab as a bottleneck, even a blockage, to the widespread use of AmP and DEBtool. (As retiree, I don't have institutional access to Matlab, and don't see myself ever purchasing my own copy).

So rather than ad hoc tinkering for one lecture on my course, seems to me it would be better to modify the code in the main DynamicEnergyBudgets package so that it is easy to run simulations --- say completely arbitrary parameters, using AmP fitted parameters, even interfacing with AmP? With the ambition eventually to provide a complete alternative to DEBtool?

Not sure whether we should discuss this here, I could email you privately?

**Comment by rafaqz** [comment]
It's interesting you want to use DynamicEnergyBudgets.jl rather than DEBtool. I felt the same about matlab. I could help out a little with that, but not likely in the next month as I also have some teaching coming up with other packages. Feel free to tinker and make PRs to the package if you wan to change anything.

Some of the code is overly complicated because I was using it to do live visualisations so I could understand the parameters of the plant model for that paper. I needed to track all the internal variables during the simulation. I should also clarify that I wrote these packages in my first year or two using Julia - I still think its a good proof of concept, but there are some design mistakes. You will probably find some things broken with DynamicEnergyBudgets.jl.

If using these packages in something people want I will consider putting a little more maintenance into them when my other packages are demanding less of my time.

**Comment by HenriLaurie** [comment]
OK, it seems best then to do the following in the short term: add an animal
type (initially just isomorph, standard DEB model---including at least one
reproduction type eg. eggs, ageing i.e. probability of death) and a few
ways to generate results (size, reproduction, oxygen consumption etc),
together with some calling interface. First of all my own fork, I am not
very confident I can write sufficiently high quality code. And as to design
mistakes ... out of my league. If this goes OK, PRs then follow.

I am sure we would find several potential collaborators for a DEB
Matlab-to-Julia migration, it would so speed up AmP, and make further
collaboration so much easier! In this respect, the fact that the two
databases exist as .mat files frees up development nicely. Of course
replacing Matlab with Julia would be the distant goal, but it need not be
an immediate objective.

So approximately November then!

**Comment by rafaqz** [comment]
Yes adding an `Animal` type is a good idea. You can give it defaults with a single structure, rather than 2.

**Comment by HenriLaurie** [comment]
So I have started another code body as a personal repo:
https://github.com/HenriLaurie/DEBStandardModel. I wasn't able to get  your
model working, though perhaps I should try again under v1.7.

In any case, I wanted to make available my alternative to Add-my-Pet's
approach to obtaining length at birth. I believe using  the scaled
equations for (uE, l, uH), see page 55 of DEB book 3rd ed, and the callback
mechanism of DifferentialEquations is the way to go (I think this provides
the entire standard model, but admittedly I haven't looked into using it to
get respiration rate and actual chemical composition, nor have I given
thought to extensions).

I am really not sure what the way forward is. If we want a systematic
replacement of Matlab in Add-my-Pet, it seems to me the main issue is the
data-fitting. And I guess that would mean starting with a subset of the
data, which I don't know how to construct. But perhaps you would like
immediately to adopt a design that is very general, and can take your plant
model(s) and all kinds of animal models? In that case probably we should
work on the design, and not worry too much about the parameter estimation
issues?

For my own code, the next few things are minor: plot dimensional length
versus age (= time since birth), function that solves for uE0 given uHb and
the assumption that e at birth equals f.

**Comment by rafaqz** [comment]
Sorry this totally dropped off my radar, PhD and too many other packages.

**Comment by HenriLaurie** [comment]
`Hi -- Am travelling, using borrowed laptop, but I guess you mean you don't
have to time work on replicating Add-my-Pet in Julia? -- H

**Comment by rafaqz** [comment]
Unfortunately I don't personally. But you will be pleased to hear there is a plan building to move a lot of deb and microclimate tools to Julia, which I will hope to contribute to.

**Comment by HenriLaurie** [comment]
Indeed! I look forward to that, I would also like to help, where I can.
